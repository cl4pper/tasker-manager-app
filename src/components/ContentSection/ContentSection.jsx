import React from 'react';
import classNames from 'classnames';

// STYLE
import './ContentSection.scss';

function ContentSection(props) {
	const { children, light, left, right, spaced, height, vertical } = props;
	return (
		<section
			className={classNames('ContentSection', {
				'ContentSection--dark': !light,
				'ContentSection--light': light,
				'ContentSection--center': !left && !right && !spaced,
				'ContentSection--right': right,
				'ContentSection--left': left,
				'ContentSection--spaced': spaced,
				'ContentSection--vertical': vertical && !left && !right && !spaced,
			})}
			style={{
				height: `${height}`,
			}}>
			{children}
		</section>
	);
}

export { ContentSection };
