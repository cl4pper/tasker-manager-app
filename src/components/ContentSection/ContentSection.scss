.ContentSection {
  width: 100%;

  @media (max-width: 500px) {
    padding: 0;
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  &--center {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
  }

  &--vertical {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;
  }

  &--spaced {
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    flex-wrap: wrap;
  }

  &--left {
    display: flex;
    align-items: center;
    justify-content: flex-start;
  }

  &--right {
    display: flex;
    align-items: center;
    justify-content: flex-end;
  }

  &--dark {
    background-color: $bg-color-secondary;
  }

  &--light {
    background-color: $bg-color-primary;
  }
}
