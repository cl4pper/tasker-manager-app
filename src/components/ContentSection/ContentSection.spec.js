import React from 'react';
import renderer from 'react-test-renderer';
import { ContentSection } from './ContentSection';

describe('ContentSection', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<ContentSection light={false}><p>Content</p></ContentSection>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
