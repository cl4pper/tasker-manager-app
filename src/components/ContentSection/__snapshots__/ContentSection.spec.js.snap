// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentSection snapshot renders 1`] = `
<section
  className="ContentSection ContentSection--dark ContentSection--center"
  style={
    Object {
      "height": "undefined",
    }
  }
>
  <p>
    Content
  </p>
</section>
`;
